### ISSUE TITLE
Collection in StackLayout incorrect padding and margins

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#6524]
### Description

A CollectionView was truncated unexpectedly

### Steps to Reproduce

1. Unzip the project from [simple-embeddedcollection.zip](https://github.com/dotnet/maui/files/8567997/simple-embeddedcollection.zip)
2. Compile and run it on Windows, you should see this
![simple-embeddedcollection](https://user-images.githubusercontent.com/12423046/165421168-87294b51-e0a2-4706-99ba-50d6a0e5e0e6.jpg)

It is just a layout described like this in XAML:
```
    <VerticalStackLayout Padding="40">
        <CollectionView>
            <CollectionView.ItemsSource>
                <x:Array Type="{x:Type x:String}">
                    <x:String>Item 1</x:String>
                    <x:String>Item 2</x:String>
                    <x:String>Item 3</x:String>
                </x:Array>
            </CollectionView.ItemsSource>
            <CollectionView.ItemTemplate>
                <DataTemplate>
                    <Grid Margin="50,0,50,0" ColumnDefinitions="*,Auto">
                        <Label Grid.Column="0" Text="{Binding}" FontSize="Large" />
                        <Label Grid.Column="1" Text="$123.45" FontSize="Large"  />
                    </Grid>
                </DataTemplate>
            </CollectionView.ItemTemplate>
        </CollectionView>
    </VerticalStackLayout>
```
I saw two errors:

- Setting Padding on the VerticalStackLayout caused the text in Grid column 1 to be truncated at the right margin.
- The left and right margin settings on the grid seemed to have no effect. 

The first one only seemed to happen on Windows, the second happened on Android too.  
 

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

Android 11, Windows 10 

### Did you find any workaround?

Using Margin instead of Padding and vice versa often helped work around the problem.

**Comment by v-longmin** [comment]
Verified repro on windows, not repro on Android 12.0 with VS 17.3.0 Preview 1.0 [32414.199.main]. Repro project:
[simple-embeddedcollection.zip](https://github.com/dotnet/maui/files/8568078/simple-embeddedcollection.zip)

**Comment by Redth** [comment]
Looks fixed to me in main, likely from #6762

**Comment by david-maw** [comment]
Thanks @Redth can you tag it to indicate what release I'll likely see the fix in?
